mqtt/mqtt-broker/src/sys_topic.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::HashMap;
+use std::iter::FromIterator;
+
+use mqtt3::proto;
+use tracing::error;
+
+use crate::session::Session;
+use crate::ClientId;
+
+pub enum StateChange<'a> {
+    Subscriptions(&'a ClientId, Vec<&'a str>),
+    Connections(Vec<&'a ClientId>),
```

Comment:
Could you please clarify once again why store borrowing references instead of owning values? In the end we still make a copy when deserializing.

Reply:
Vec<String> will be copied twice, once when moving into vec, and again when converting to json. Vec<&str> is only copied when converted to json